### PR TITLE
nnvp-85

### PR DESCRIPTION
**Pull request by mateusztheglue**

### CODE REVIEW

**Comment by theglue-cicd** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-45** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-45.cluster.development.nnvp.thegluesandbox.com)

File: src/main/java/com/theglue/nnvp/service/rsr/RsrEntry.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        .gender(NnvpEnum.fromCode(parseLine(line, 131, 1), Gender.class))
+        .build();
+
+  }
+
+  private static Registration parseRegistration(String line, TransactionCode transactionCode) {
+    return Registration.builder()
+        .productBranches(StreamSupport
+            .stream(Splitter.fixedLength(2).split(StringUtils.stripToEmpty(parseLine(line, 217, 10))).spliterator(), false)
+            .map(code -> Integer.valueOf(code))
```
Comment (eryk7b) [inline src/main/java/com/theglue/nnvp/service/rsr/RsrEntry.java]: Integer::valueOf

File: src/main/java/com/theglue/nnvp/service/rsr/RsrEntry.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            .map(code -> ProductBranch.builder()
+                .productBranchCode(code / 10)
+                .subBranchCode(code % 10)
+                .build())
+            .collect(Collectors.toList()))
+        .reasonCodes(StreamSupport
+            .stream(Splitter.fixedLength(2).split(StringUtils.stripToEmpty((parseLine(line, 280, 10)))).spliterator(), false)
+            .map(code -> Integer.valueOf(code))
```
Comment (eryk7b) [inline src/main/java/com/theglue/nnvp/service/rsr/RsrEntry.java]: Integer::valueOf

File: src/main/java/com/theglue/nnvp/service/rsr/RsrEntry.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    return StringUtils.isEmpty(substring) ? null : substring;
+  }
+
+  private static LocalDate parseDate(String value) {
+    if (StringUtils.isEmpty(value)) {
+      return null;
+    }
+    int day = Integer.valueOf(value.substring(0, 2));
+    int month = Integer.valueOf(value.substring(2, 4));
+    int year = Integer.valueOf(value.substring(4, 8));
```
Comment (eryk7b) [inline src/main/java/com/theglue/nnvp/service/rsr/RsrEntry.java]: Integer.parseInt

File: src/main/resources/db/migration/V1_1_5__create_job_info.sql [hunk 4]
```diff
@@ -0,0 +1,7 @@
+create table job_info
+(
+    id                  int(10) NOT NULL AUTO_INCREMENT,
+    job_type            varchar(20),
+    job_status          varchar(20),
+    PRIMARY KEY (id)
+);
```
Comment (eryk7b) [inline src/main/resources/db/migration/V1_1_5__create_job_info.sql]: new line

**Comment by theglue-cicd** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-45** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-45.cluster.development.nnvp.thegluesandbox.com)

**Pull request merged**